src/test/java/io/jenkins/plugins/ml/IPythonUserConfigTest.java

```diff
@@ -0,0 +1,41 @@
+package io.jenkins.plugins.ml;
+
+import org.apache.zeppelin.interpreter.InterpreterException;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class IPythonUserConfigTest {
+
+    private IPythonUserConfig userConfig;
+
+    @Before
+    public void setup() {
+        // setup configuration
+        userConfig = new IPythonUserConfig();
+        userConfig.setServerGatewayAddress("127.0.0.1");
+        userConfig.setIPythonLaunchTimeout(2000);
+        userConfig.setMaxResult(2);
+    }
+
+    @Test
+    public void testDefaultIPythonConfig() throws InterpreterException, IOException {
+        userConfig = new IPythonUserConfig();
+        assertEquals("Server address not matched","127.0.0.1",userConfig.getServerGatewayAddress());
+        assertEquals("Timeout not matched",1000,userConfig.getIPythonLaunchTimeout());
+        assertEquals("Max results not matched",3,userConfig.getMaxResult());
+
+        // setup Interpreter
+        InterpreterManager interpreterManager = new IPythonInterpreterManager(userConfig);
+        interpreterManager.initiateInterpreter();
+
+        //Check whether it runs or not
+        String result = interpreterManager.invokeInterpreter("print('default')");
```

Comment:
The reason behind this `testDefaultIPythonConfig – io.jenkins.plugins.ml.IPythonUserConfigTest` failure is the server behind Jenkins build have not met prerequisites for connecting to an IPython kernel. Should I remove the test or comment? If we will be able to setup ipython env in the future, we may have the test later .

Reply:
If the test needs a live instance of IPython running, that might be a bit difficult.

Easier to mock it I think.

Reply:
If we don't have running an IPython instance, we can write tests barely for the integration tests. Do we have any of a chance/procedure to request?

Reply:
I don't think we can request others to set up and maintain a server with IPython, it that's what you are asking. For unit tests, we can probably mock. Or we can use a Docker container during build to start an instance that is destroyed after the tests (though I only know how to do that with Travis & GH actions, but I believe Jenkins infra should have something similar)

Reply:
Yes, that was on my mind. We can now focus on the implementation, later we can add docker and travis configs.